lib/services/unified_resource.go

```diff
@@ -670,7 +701,20 @@ func (c *UnifiedResourceCache) processEventsAndUpdateCurrent(ctx context.Context
 		case types.OpDelete:
 			c.deleteLocked(event.Resource)
 		case types.OpPut:
-			c.putLocked(event.Resource.(resource))
+			r := event.Resource
+			if u, ok := r.(types.Resource153Unwrapper); ok {
+				switch unwrapped := u.Unwrap().(type) {
+				case IdentityCenterAccount:
+					r = WrapUnifiedResource153(unwrapped)
```

Comment:
This is confusing. Why do we need to unwrap and then wrap it back? Is there a cleaner way to do this?